build_wheels.py

```diff
@@ -0,0 +1,113 @@
+import json
+
+import requests
+from packaging.requirements import InvalidRequirement
+from packaging.requirements import Requirement
+
+idf_constraints_url = 'https://dl.espressif.com/dl/esp-idf/espidf.constraints.'
+
+# --- Global variables for branches and constraints files to be downloaded requirements for --- #
+# IDF branches are automatically fetched from github, every release branch higher or equal than 5.0 and master is taken
+# For constraints file the version is resolved from IDF release branch
+# The constraints file for master branch is considered to be the latest release version
+# incremented by 1 in minor field (e.g when last release is v5.2 the constrains file for master is v5.3)
+# if there is no such a version, the last release branch constraints file will be used
+# This can be overwritten by setting the variable 'AUTO_VERSION_CONSTRAINS_FILE' to False
+# and append into the list (in if statement bellow) desired version for master branch
+
+AUTO_VERSION_CONSTRAINS_FILE = True
+
+# ESP-IDF branches list
+IDF_BRANCHES_URL = 'https://api.github.com/repos/espressif/esp-idf/branches?protected=true&per_page=100'
+r = requests.get(IDF_BRANCHES_URL)
+if r.status_code == 200:
+    idf_repo_branches = [branch['name'] for branch in r.json()]
+else:
+    print('Failed to fetch ESP-IDF branches.', '\n', r.text)
+    exit()
+
+idf_branches = []
+for branch in idf_repo_branches:
+    if branch.startswith('release'):
+        if float(branch.split('v')[1]) >= 5.0:
+            idf_branches.append(branch)
+idf_branches.append('master')
+
+# Constraints files versions list
+idf_constraints = []
+for branch in idf_branches:
+    try:
+        splitted_branch = branch.split('release/')
+        idf_constraints.append(splitted_branch[1])
+    except IndexError:
+        if AUTO_VERSION_CONSTRAINS_FILE:
+            idf_last_release = idf_constraints[-1].split('v')[1].split('.')
+            idf_master_minor = str(int(idf_last_release[1])+1)
+            if requests.get(f'{idf_constraints_url}v{idf_last_release[0]}.{idf_master_minor}.txt').status_code == 200:
+                idf_constraints.append(f'v{idf_last_release[0]}.{idf_master_minor}')
+            else:
+                idf_constraints.append(idf_constraints[-1])
+        else:
+            idf_constraints.append('6.0')
```

Comment:
What is this hard-coded 6.0?

Reply:
This is in case the `AUTO_VERSION_CONSTRAINS_FILE` will be `false`. I was thinking if the last IDF release version will be `v5.X` the auto version will try to assign for master branch `v5.(X+1)` but there will be the major release planned and master will be `v6.0`.

Yes, there is an fallback option for last release constraints file but I thought it could be needed to match master with the right constraints file.

But I should leave a comment in code for this, thanks.